src/args.rs

```diff
@@ -0,0 +1,47 @@
+#[derive(Debug, PartialEq, Eq)]
+pub struct Args {
+    pub command: String
+}
+
+impl Args {
+    pub fn parse(raw_args: Vec<String>) -> Result<Args, String> {
+        match raw_args.len() {
+            0 => Err(String::from("Please pass remote command.")), // TODO more user friendly message, for now it's consistent with Bash version.
+            _ => Ok(Args {
+                command: {
+                    let str: String = raw_args
+                        .iter()
+                        .cloned()
+                        // Add space between parameters.
+                        .map(|arg| format!("{} ", arg))
```

Comment:
I guess, this isn't important, but this code will allocate a new String for result. This is sorta wasteful given that you already have a owned `String` that you can modify here.